app/src/composables/use-item/index.ts

```diff
@@ -86,9 +87,33 @@ export function useItem<T extends Item>(
 		return `${getEndpoint(collection.value)}/${encodeURIComponent(primaryKey.value as string)}`;
 	});
 
-	const defaultValues = getDefaultValuesFromFields(fieldsWithPermissions);
+	const defaultValues = computed(() => {
+		const schemaDefaults = getDefaultValuesFromFields(fieldsWithPermissions).value;
 
-	watch([collection, primaryKey, ...(isRef(query) ? [query] : [])], refresh);
+		// For new items, merge in prefill values from URL parameters
+		if (isNew.value) {
+			const prefill = unref(prefillValues);
```

Comment:
To be able to merge this, we would need some form of basic value parsing/validating here. Basically making sure that if the field is using a certain interface, that this interface either supports a string or that the value will be casted into whatever the interface needs. If the interface is unknown or not supported, we should ignore the default value coming from the query.